qiita_db/meta_util.py

```diff
@@ -184,12 +184,16 @@ def update_redis_stats():
     number_of_samples = {}
     ebi_samples_prep = {}
     num_samples_ebi = 0
+
     for k, sts in viewitems(studies):
         number_of_samples[k] = 0
         for s in sts:
             st = s.sample_template
             if st is not None:
-                number_of_samples[k] += len(list(st.keys()))
+                for p in s.prep_templates():
+                    if p.artifact is not None and p.artifact.visibility == k:
+                        n = len(list(p.keys()))
```

Comment:
I _think_ the easiest to accomplish the goal of having the most accurate count is to count each status per study via this prep loop and then just add those counts to the total count ... something like this:
```python
if st is not None:
    prep_totals = {'public': 0, 'sandbox': 0, 'private': 0, 'awaiting_approval': 0}
    for p in s.prep_templates():
           if p.artifact is not None and p.artifact is not None:
                prep_totals[p.artifact.visibility] += len(list(p.keys()))
   ... 
```

Reply:
Unfortunately this would still over count when a sample is on multiple preps